### PR TITLE
Adds reporting of macaddress interfaces to bcdb

### LINKED ISSUE #591: Report mac addresses of interfaces to bcdb

### PR DESCRIPTION
**Pull request by DylanVerstraete**

### CODE REVIEW

**Comment by DylanVerstraete** [comment]
@zaibon @muhamadazmy is there anything missing to finish this?

**Review by zaibon — COMMENTED** [review]
This look good.
Only one question. does the type `net.HardwareAddr` serialize find in json ?

File: tools/bcdb_mock/models/generated/directory/directory.go [hunk 1]
```diff
@@ -96,9 +96,10 @@ func NewTfgridDirectoryNode2() (TfgridDirectoryNode2, error) {
 }
 
 type TfgridDirectoryNodeIface1 struct {
-	Name    string           `bson:"name" json:"name"`
-	Addrs   []schema.IPRange `bson:"addrs" json:"addrs"`
-	Gateway []net.IP         `bson:"gateway" json:"gateway"`
+	Name       string           `bson:"name" json:"name"`
+	Addrs      []schema.IPRange `bson:"addrs" json:"addrs"`
+	Gateway    []net.IP         `bson:"gateway" json:"gateway"`
+	MacAddress net.HardwareAddr `bson:"macaddress" json:"macaddress"`
```
Comment (muhamadazmy) [inline tools/bcdb_mock/models/generated/directory/directory.go]: You need to add the corresponding field in this file https://github.com/threefoldtech/zos/blob/master/tools/bcdb_mock/models/schema/directory/tfgrid_directory_node_2.toml#L31

You probably need to add something like
```
macaddress = (S) 
```

File: pkg/network/types/types.go [hunk 2]
```diff
@@ -22,9 +22,10 @@ const (
 // that the node will publish publicly
 // this is used to be able to configure public side of a node
 type IfaceInfo struct {
-	Name    string   `json:"name"`
-	Addrs   []IPNet  `json:"addrs"`
-	Gateway []net.IP `json:"gateway"`
+	Name       string           `json:"name"`
+	Addrs      []IPNet          `json:"addrs"`
+	Gateway    []net.IP         `json:"gateway"`
+	MacAddress net.HardwareAddr `json:"macaddress"`
```
Comment (LeeSmet) [inline pkg/network/types/types.go]: net.HardwareAddr does not have marshal methods afaik, so are we sure this is going to properly print in json? (`xx:xx:xx:xx:xx:xx`)?

File: pkg/network/types/types_test.go [hunk 3]
```diff
@@ -86,6 +86,36 @@ func TestDumpIPNet(t *testing.T) {
 	}
 }
 
+func TestDumpMacaddress(t *testing.T) {
+	mustParse := func(in string) schema.MacAddress {
+		mac, err := net.ParseMAC(in)
+		if err != nil {
```
Comment (zaibon) [inline pkg/network/types/types_test.go]: this is just a test, no need to panic
```suggestion
              require.NoError(t,err)
```

**Review by zaibon — COMMENTED** [review]
Very nice

**Comment by DylanVerstraete** [comment]
Don't merge yet, bcdb mock is throwing an error. Need to investigate

edit: works now! :ship: it

**Review by zaibon — APPROVED** [review]

**Review by zaibon — APPROVED** [review]

**Pull request merged**